src/main/java/org/jabref/model/study/FetchResult.java

```diff
@@ -0,0 +1,24 @@
+package org.jabref.model.study;
+
+import org.jabref.model.database.BibDatabase;
+
+/**
+ * Helper class used to make crawling in study fetcher possible with streams.
+ */
+public class FetchResult {
+    private final String fetcher;
+    private final BibDatabase fetchResult;
+
+    public FetchResult(String fetcher, BibDatabase fetcherResult) {
```

Comment:
I think, `Fetcher` exists as class - to avoid confusion

```suggestion
    public FetchResult(String fetcherName, BibDatabase fetcherResult) {
```